org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMProcessingInstructionFormatter.java

```diff
@@ -38,6 +38,12 @@ public DOMProcessingInstructionFormatter(XMLFormatterDocumentNew formatterDocume
 	public void formatProcessingInstruction(DOMProcessingInstruction processingInstruction,
 			XMLFormattingConstraints parentConstraints, List<TextEdit> edits) {
 		int prevOffset = processingInstruction.getStartContent();
+		if (processingInstruction.getParentElement() != null) {

```

Comment:
Define a variable

`DOMElement parentElement = processingInstruction.getParentElement() ` and use it for the following code.